search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -70,18 +75,25 @@
   (map #(rfm->umm-collection context %) rfms))
 
 (defn- wait-and-retry-for-collection-cache
-  "Wait configurable number of seconds before retrying to get all the collections
-   from collection cache. If collections are still not present, throw warning."
+  "Wait configurable number of milli seconds before retrying configurable number of times
+   to get all the collections from collection cache. If collections are still not present, throw warning."
   [context]
   (info (format "Collection cache is not populated after %d seconds of delay" 
                 (humanizer-report-generator-job-delay)))
-  (Thread/sleep (humanizer-report-generator-job-wait)) 
+  ;; sleep, retry 19 times, then sleep and retry one last time.
+  (Thread/sleep (humanizer-report-generator-job-wait))
+  (let [a (atom (- (humanizer-report-generator-job-retry-number) 1))]
```

Comment:
I would just name a something a little more descriptive like atom-retry-count. You could also call `(dec (humanizer-report-generator-job-retry-number))` here.